File: A10 Recursion, DFS, Backtracking, BFS/102 Binary Tree LevelOrder Traversal/DFS.py
# DFS recursively
def levelOrder1(self, root):
    res = []
    self.dfs(root, 0, res)
    return res
    
def dfs(self, root, level, res):
    if not root:
        return 
    if len(res) < level+1:
        res.append([])
    res[level].append(root.val)
    self.dfs(root.left, level+1, res)
    self.dfs(root.right, level+1, res)

File: A06 Nodes Related/002 Binary Tree/LC107 Binary Tree Level Order Traversal II/DFS.java
/*
Time Complexity:
Space Complexity:

*/ 

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>(); 
        Queue<TreeNode> queue = new LinkedList<>(); 
        
        if (root == null) return res;
        
        queue.offer(root);
        while(!queue.isEmpty()){
            int size = queue.size();
            List<Integer> sublist = new ArrayList<>(); 
            for(int i = 0 ; i < size; i++){
                TreeNode temp = queue.poll();
                sublist.add(temp.val);
                if (temp.left != null) queue.offer(temp.left); 
                if (temp.right != null) queue.offer(temp.right);
            }
            res.add(0, sublist);
        }
        return res;
    }
}